src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -251,6 +276,41 @@ private static bool ValidateRestoreGraphs(IEnumerable<RestoreTargetGraph> graphs
                     }
                 };
 
+                // Try to find the existing lock file. Since the existing lock file is pathed under
+                // a folder that includes the resolved tool's version, this is a bit of a chicken
+                // and egg problem. That is, we need to run the restore operation in order to resolve
+                // a tool version, but we need the tool version to find the existing project.lock.json
+                // file which is required before executing the restore! Fortunately, this is solved by
+                // looking at the tool's consuming project's lock file to see if the tool has been
+                // restored before.
+                LockFile existingToolLockFile = null;
+                if (_request.ExistingLockFile != null)
+                {
+                    var existingTarget = _request
+                        .ExistingLockFile
+                        .Tools
+                        .Where(t => t.RuntimeIdentifier == null)
+                        .Where(t => t.TargetFramework.Equals(LockFile.ToolFramework))
+                        .FirstOrDefault();
+
+                    var existingLibrary = existingTarget?.Libraries
+                        .Where(l => StringComparer.OrdinalIgnoreCase.Equals(l.Name, tool.LibraryRange.Name))
+                        .Where(l => tool.LibraryRange.VersionRange.Satisfies(l.Version))
+                        .FirstOrDefault();
+
+                    if (existingLibrary != null)
+                    {
+                        var existingLockFilePath = toolPathResolver.GetLockFilePath(
+                            existingLibrary.Name,
+                            existingLibrary.Version,
+                            existingTarget.TargetFramework);
+
+                        existingToolLockFile = LockFileUtilities.GetLockFile(existingLockFilePath, _logger);
+                    }
+                }
+
+                var relockFile = ShouldRelockFile(existingToolLockFile, toolPackageSpec);
```

Comment:
Since the lock file format supports it, yes, the code is in place. However, I would like to understand more about what scenario sets this property to `true`.